redis/src/main/scala/zio/redis/options/Connection.scala

```diff
@@ -82,11 +148,269 @@ trait Connection {
     events: ClientEvents = ClientEvents(),
     lastCommand: Option[String] = None,
     argvMemory: Option[Long] = None,
+    multiMemory: Option[Long] = None,
     totalMemory: Option[Long] = None,
+    replyBufferSize: Option[Long] = None,
+    replyBufferPeakPosition: Option[Long] = None,
     redirectionClientId: Option[Long] = None,
+    respProtocolVersion: Option[String] = None,
     user: Option[String] = None
   )
 
+  object ClientInfoCodec extends zio.schema.codec.Codec {
+    override def encoder[A](schema: Schema[A]): ZTransducer[Any, Nothing, A, Byte] =
+      ZTransducer.fromPush((opt: Option[Chunk[A]]) =>
+        ZIO
+          .effect(opt.map(values => values.flatMap(encode(schema))).getOrElse(Chunk.empty))
+          .orDie
+      )
+
+    override def decoder[A](schema: Schema[A]): ZTransducer[Any, String, Byte, A] =
+      ZTransducer.utfDecode >>> ZTransducer.fromFunctionM((s: String) =>
```

Comment:
Need to sync master branch , update with zio2.

Reply:
merged with master and rebased